caikit/runtime/service_factory.py

```diff
@@ -311,9 +286,8 @@ def get_train_request(module_class: Type[ModuleBase]) -> Type[DataBase]:
         module_class,
         ModuleBase,
     )
-    request_class_name = (
-        f"{module_class.TASK_CLASS.__name__}{module_class.__name__}TrainRequest"
-    )
+    first_task = next(iter(module_class.tasks))
+    request_class_name = f"{first_task.__name__}{module_class.__name__}TrainRequest"
```

Comment:
Ah, I think I skimmed past this on first review and now I understand your question in the PR body better. I don't have any great ideas on how to name the training rpc now that a module may have multiple tasks. We'll probably need to address that soon-ish, but this is fine to go with for now.

This also highlights the extra coupling between this logic and the rpc implementation- any change in the naming scheme over in `rpc.py` has to be reflected here.
There's probably a better way to get at the reference to the `ModuleClassTrainRPC` reference and return the request class from that, but that doesn't need to be fixed in this PR.